### ISSUE TITLE
Snapshot failed to backup to Google Drive

### ISSUE THREAD

**Issue by MisterTFM** [sabeechen/hassio-google-drive-backup#343]
Copy the info from the previou
###### Description:
```
If you have anything else that could help explain what happened, click "Markdown" above and write it here.
```

 Addon version: `0.103.1`
 Home Assistant Version: `2021.1.5`
 Supervisor Version: `2021.01.7`
 Supervisor Channel: `stable`
 Hassos Version: `5.10`
 Docker Version: `19.03.13`
 Architecture: `armv7`
 Machine: `raspberrypi4`
 Date: `2021-02-08T13:39:12.738906+00:00`
 Timezone: `America/New_York`
 Failure Time: `2021-02-08T13:24:27.962795+00:00`
 Last Good Sync: `2021-02-08T12:47:54.235182+00:00`
 ###### Exception:
 ```
 
  addon/backup/model/coordinator.py:156 (_sync)
  addon/backup/model/model.py:180 (sync)
  addon/backup/drive/drivesource.py:173 (save)
  addon/backup/drive/drivesource.py:160 (save)
  addon/backup/drive/driverequests.py:280 (create)
  addon/backup/drive/driverequests.py:265 (create)
  addon/backup/drive/driverequests.py:330 (retryRequest)
  addon/backup/creds/driverequester.py:39 (request)
  /site-packages/aiohttp/client_reqrep.py:1000 (raise_for_status)
aiohttp.client_exceptions.ClientResponseError: 408, message='Request Timeout', url=URL('https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UyAJ932A-nKEe-ebBQNv9WAxrwk-4VyHLYCd42jpCfyMvdFN7eM8tvblMWETHNSkxnYE0jTGw6FY9uxeuDD2_s')

 ```
 Snapshots:
 ```
 {
    "HomeAssistant": {
        "snapshots": 2,
        "retained": 0,
        "deletable": 2,
        "name": "HomeAssistant",
        "title": "Home Assistant",
        "latest": "2021-02-08T02:35:19Z",
        "max": 2,
        "enabled": true,
        "size": "4.7 GB",
        "free_space": "13.0 GB"
    },
    "GoogleDrive": {
        "snapshots": 2,
        "retained": 0,
        "deletable": 2,
        "name": "GoogleDrive",
        "title": "Google Drive",
        "latest": "2021-02-06T19:04:17Z",
        "max": 2,
        "enabled": true,
        "size": "4.6 GB"
    }
}
 ```
 ###### Config:
 ```
 {
    "Setting.MAX_SNAPSHOTS_IN_HASSIO": 2,
    "Setting.MAX_SNAPSHOTS_IN_GOOGLE_DRIVE": 2,
    "Setting.DAYS_BETWEEN_SNAPSHOTS": 2.0,
    "Setting.SNAPSHOT_TIME_OF_DAY": "07:00",
    "Setting.NOTIFY_FOR_STALE_SNAPSHOTS": false,
    "Setting.SEND_ERROR_REPORTS": true,
    "Setting.GOOGLE_DRIVE_TIMEOUT_SECONDS": 480.0
}
 ```
 ###### Addon Logs:
 ```
 02-08 08:16:55 DEBUG [backup.drive.drivesource] Uploading Full Snapshot 2021-02-07 21:35:19 25.47%
02-08 08:16:55 DEBUG [backup.drive.driverequests] Sending 1048576 bytes to Google Drive
02-08 08:16:55 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UyAJ932A-nKEe-ebBQNv9WAxrwk-4VyHLYCd42jpCfyMvdFN7eM8tvblMWETHNSkxnYE0jTGw6FY9uxeuDD2_s
02-08 08:16:59 DEBUG [backup.drive.drivesource] Uploading Full Snapshot 2021-02-07 21:35:19 25.51%
02-08 08:16:59 DEBUG [backup.drive.driverequests] Sending 2097152 bytes to Google Drive
02-08 08:16:59 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UyAJ932A-nKEe-ebBQNv9WAxrwk-4VyHLYCd42jpCfyMvdFN7eM8tvblMWETHNSkxnYE0jTGw6FY9uxeuDD2_s
02-08 08:17:05 DEBUG [backup.drive.drivesource] Uploading Full Snapshot 2021-02-07 21:35:19 25.60%
02-08 08:17:05 DEBUG [backup.drive.driverequests] Sending 3407872 bytes to Google Drive
02-08 08:17:05 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UyAJ932A-nKEe-ebBQNv9WAxrwk-4VyHLYCd42jpCfyMvdFN7eM8tvblMWETHNSkxnYE0jTGw6FY9uxeuDD2_s
02-08 08:17:25 DEBUG [backup.drive.drivesource] Uploading Full Snapshot 2021-02-07 21:35:19 25.73%
02-08 08:17:25 DEBUG [backup.drive.driverequests] Sending 1572864 bytes to Google Drive
02-08 08:17:25 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UyAJ932A-nKEe-ebBQNv9WAxrwk-4VyHLYCd42jpCfyMvdFN7eM8tvblMWETHNSkxnYE0jTGw6FY9uxeuDD2_s
02-08 08:17:36 DEBUG [backup.drive.drivesource] Uploading Full Snapshot 2021-02-07 21:35:19 25.80%
02-08 08:17:36 DEBUG [backup.drive.driverequests] Sending 1310720 bytes to Google Drive
02-08 08:17:36 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UyAJ932A-nKEe-ebBQNv9WAxrwk-4VyHLYCd42jpCfyMvdFN7eM8tvblMWETHNSkxnYE0jTGw6FY9uxeuDD2_s
02-08 08:17:45 DEBUG [backup.drive.drivesource] Uploading Full Snapshot 2021-02-07 21:35:19 25.85%
02-08 08:17:45 DEBUG [backup.drive.driverequests] Sending 1310720 bytes to Google Drive
02-08 08:17:45 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UyAJ932A-nKEe-ebBQNv9WAxrwk-4VyHLYCd42jpCfyMvdFN7eM8tvblMWETHNSkxnYE0jTGw6FY9uxeuDD2_s
02-08 08:17:54 DEBUG [backup.drive.drivesource] Uploading Full Snapshot 2021-02-07 21:35:19 25.90%
02-08 08:17:54 DEBUG [backup.drive.driverequests] Sending 1310720 bytes to Google Drive
02-08 08:17:54 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UyAJ932A-nKEe-ebBQNv9WAxrwk-4VyHLYCd42jpCfyMvdFN7eM8tvblMWETHNSkxnYE0jTGw6FY9uxeuDD2_s
02-08 08:18:13 DEBUG [backup.drive.drivesource] Uploading Full Snapshot 2021-02-07 21:35:19 25.95%
02-08 08:18:13 DEBUG [backup.drive.driverequests] Sending 524288 bytes to Google Drive
02-08 08:18:13 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UyAJ932A-nKEe-ebBQNv9WAxrwk-4VyHLYCd42jpCfyMvdFN7eM8tvblMWETHNSkxnYE0jTGw6FY9uxeuDD2_s
02-08 08:18:15 DEBUG [backup.drive.drivesource] Uploading Full Snapshot 2021-02-07 21:35:19 25.98%
02-08 08:18:15 DEBUG [backup.drive.driverequests] Sending 3145728 bytes to Google Drive
02-08 08:18:15 DEBUG [backup.drive.driverequests] Making Google Drive request: https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UyAJ932A-nKEe-ebBQNv9WAxrwk-4VyHLYCd42jpCfyMvdFN7eM8tvblMWETHNSkxnYE0jTGw6FY9uxeuDD2_s
02-08 08:24:27 ERROR [backup.model.coordinator] 
  addon/backup/model/coordinator.py:156 (_sync)
  addon/backup/model/model.py:180 (sync)
  addon/backup/drive/drivesource.py:173 (save)
  addon/backup/drive/drivesource.py:160 (save)
  addon/backup/drive/driverequests.py:280 (create)
  addon/backup/drive/driverequests.py:265 (create)
  addon/backup/drive/driverequests.py:330 (retryRequest)
  addon/backup/creds/driverequester.py:39 (request)
  /site-packages/aiohttp/client_reqrep.py:1000 (raise_for_status)
aiohttp.client_exceptions.ClientResponseError: 408, message='Request Timeout', url=URL('https://www.googleapis.com/upload/drive/v3/files/?uploadType=resumable&supportsAllDrives=true&upload_id=ABg5-UyAJ932A-nKEe-ebBQNv9WAxrwk-4VyHLYCd42jpCfyMvdFN7eM8tvblMWETHNSkxnYE0jTGw6FY9uxeuDD2_s')

02-08 08:24:27 INFO [backup.model.coordinator] I'll try again in an hour
02-08 08:24:29 INFO [backup.debugworker] Sending error report (see settings to disable)
 ```
 ###### Supervisor Logs:
 ```
 [32m21-02-08 12:21:10 INFO (MainThread) [supervisor.api.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-08 12:22:11 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-08 12:22:17 INFO (MainThread) [supervisor.api.security] /snapshots/a1c297ec/download access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-08 12:22:17 INFO (MainThread) [supervisor.api.snapshots] Downloading snapshot a1c297ec[0m
[32m21-02-08 12:22:17 INFO (MainThread) [supervisor.api.security] /snapshots/a1c297ec/download access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-08 12:22:17 INFO (MainThread) [supervisor.api.snapshots] Downloading snapshot a1c297ec[0m
[32m21-02-08 12:29:32 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m21-02-08 12:44:20 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-08 12:47:29 INFO (MainThread) [supervisor.api.security] /core/info access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-08 12:47:29 INFO (MainThread) [supervisor.api.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-08 12:47:54 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-08 12:48:00 INFO (MainThread) [supervisor.api.security] /snapshots/a1c297ec/download access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-08 12:48:00 INFO (MainThread) [supervisor.api.snapshots] Downloading snapshot a1c297ec[0m
[32m21-02-08 12:48:00 INFO (MainThread) [supervisor.api.security] /snapshots/a1c297ec/download access from cebe7a76_hassio_google_drive_backup[0m
[32m21-02-08 12:48:00 INFO (MainThread) [supervisor.api.snapshots] Downloading snapshot a1c297ec[0m
[32m21-02-08 12:57:39 INFO (SyncWorker_3) [supervisor.docker.interface] Stopping addon_core_samba application[0m
[32m21-02-08 12:57:44 INFO (SyncWorker_3) [supervisor.docker.interface] Cleaning addon_core_samba application[0m
[32m21-02-08 12:57:45 INFO (SyncWorker_2) [supervisor.docker.addon] Starting Docker add-on homeassistant/armv7-addon-samba with version 9.3.0[0m
[32m21-02-08 12:58:38 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.RUNNING[0m
[32m21-02-08 12:58:38 INFO (MainThread) [supervisor.resolution.check] System checks complete[0m
[32m21-02-08 12:59:33 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m21-02-08 13:00:35 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/stable.json[0m
[32m21-02-08 13:07:15 INFO (MainThread) [supervisor.host.info] Updating local host information[0m
[32m21-02-08 13:07:15 INFO (MainThread) [supervisor.host.services] Updating service information[0m
[32m21-02-08 13:07:15 INFO (MainThread) [supervisor.host.network] Updating local network information[0m
[32m21-02-08 13:07:19 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information[0m
[32m21-02-08 13:07:19 INFO (MainThread) [supervisor.host] Host information reload completed[0m
[32m21-02-08 13:29:40 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token[0m
[32m21-02-08 13:39:12 INFO (MainThread) [supervisor.api.security] /supervisor/logs access from cebe7a76_hassio_google_drive_backup[0m

 ```
 ###### Home Assistant Core Logs:
 ```
 [33m2021-02-08 08:33:20 WARNING (MainThread) [homeassistant.components.pi_hole] Failed to connect: Can not load data from *hole: 192.168.1.145:80[0m
[33m2021-02-08 08:33:20 WARNING (MainThread) [homeassistant.config_entries] Config entry for pi_hole not ready yet. Retrying in 80 seconds[0m
cat: can't open '/sys/class/thermal/thermal_zone2/temp': No such file or directory
[31m2021-02-08 08:34:17 ERROR (SyncWorker_8) [homeassistant.components.command_line] Command failed: cat /sys/class/thermal/thermal_zone2/temp[0m
[33m2021-02-08 08:34:33 WARNING (MainThread) [homeassistant.config_entries] Config entry for ipp not ready yet. Retrying in 80 seconds[0m
[31m2021-02-08 08:34:41 ERROR (MainThread) [hole] Can not load data from *hole: 192.168.1.145:80[0m
[33m2021-02-08 08:34:41 WARNING (MainThread) [homeassistant.components.pi_hole] Failed to connect: Can not load data from *hole: 192.168.1.145:80[0m
[33m2021-02-08 08:34:41 WARNING (MainThread) [homeassistant.config_entries] Config entry for pi_hole not ready yet. Retrying in 80 seconds[0m
cat: can't open '/sys/class/thermal/thermal_zone2/temp': No such file or directory
[31m2021-02-08 08:35:17 ERROR (SyncWorker_7) [homeassistant.components.command_line] Command failed: cat /sys/class/thermal/thermal_zone2/temp[0m
[33m2021-02-08 08:35:56 WARNING (MainThread) [homeassistant.config_entries] Config entry for ipp not ready yet. Retrying in 80 seconds[0m
[31m2021-02-08 08:36:04 ERROR (MainThread) [hole] Can not load data from *hole: 192.168.1.145:80[0m
[33m2021-02-08 08:36:04 WARNING (MainThread) [homeassistant.components.pi_hole] Failed to connect: Can not load data from *hole: 192.168.1.145:80[0m
[33m2021-02-08 08:36:04 WARNING (MainThread) [homeassistant.config_entries] Config entry for pi_hole not ready yet. Retrying in 80 seconds[0m
cat: can't open '/sys/class/thermal/thermal_zone2/temp': No such file or directory
[31m2021-02-08 08:36:17 ERROR (SyncWorker_1) [homeassistant.components.command_line] Command failed: cat /sys/class/thermal/thermal_zone2/temp[0m
cat: can't open '/sys/class/thermal/thermal_zone2/temp': No such file or directory
[31m2021-02-08 08:37:17 ERROR (SyncWorker_0) [homeassistant.components.command_line] Command failed: cat /sys/class/thermal/thermal_zone2/temp[0m
[33m2021-02-08 08:37:19 WARNING (MainThread) [homeassistant.config_entries] Config entry for ipp not ready yet. Retrying in 80 seconds[0m
[31m2021-02-08 08:37:27 ERROR (MainThread) [hole] Can not load data from *hole: 192.168.1.145:80[0m
[33m2021-02-08 08:37:27 WARNING (MainThread) [homeassistant.components.pi_hole] Failed to connect: Can not load data from *hole: 192.168.1.145:80[0m
[33m2021-02-08 08:37:27 WARNING (MainThread) [homeassistant.config_entries] Config entry for pi_hole not ready yet. Retrying in 80 seconds[0m
cat: can't open '/sys/class/thermal/thermal_zone2/temp': No such file or directory
[31m2021-02-08 08:38:17 ERROR (SyncWorker_6) [homeassistant.components.command_line] Command failed: cat /sys/class/thermal/thermal_zone2/temp[0m
[33m2021-02-08 08:38:42 WARNING (MainThread) [homeassistant.config_entries] Config entry for ipp not ready yet. Retrying in 80 seconds[0m
[31m2021-02-08 08:38:49 ERROR (MainThread) [homeassistant.components.upnp] Timeout fetching tfm103 data[0m
[31m2021-02-08 08:38:50 ERROR (MainThread) [hole] Can not load data from *hole: 192.168.1.145:80[0m
[33m2021-02-08 08:38:50 WARNING (MainThread) [homeassistant.components.pi_hole] Failed to connect: Can not load data from *hole: 192.168.1.145:80[0m
[33m2021-02-08 08:38:50 WARNING (MainThread) [homeassistant.config_entries] Config entry for pi_hole not ready yet. Retrying in 80 seconds[0m

 ```
 s page here

**Comment by sabeechen** [comment]
Does the backup resume where it left off after this?  Or does it just keep happening and never complete?  HTTP error 408 means the request timed out and it should rety, but if it happens often enough and your snapshot is very large the addon might give up and start over again and again preventing it from ever uploading completely.

**Comment by MisterTFM** [comment]
Sorry for the delay, I wanted to wait and see if my last backup attempt finished first, before I reported my findings.

My HA snapshots are 2.5 gb.  When the backups to Google Drive failed, they would restart on their own, but always start back from the beginning at 0%.

I changed hardware a couple days ago, then this error started occurring.  I unknowingly installed a bad NetGear ethernet switch to increase the number of ethernet ports.  There must have been a lot of colliding packets or a long delay going on, because the uploads keep failing.  I even had increased the Google drive Timout up to 480, but it kept failing.  The highest I got was 65%.

This evening I bought a new switch, restarted the backup, and it successfully uploaded to Google Drive.

It would be nice if it could restart where it failed at, but with large snapshots, that could be a problem coding.

Thanks for your reply.

Tom

________________________________

**Comment by sabeechen** [comment]
Ah, glad you could sort it out.  The addon will start back up where it left off for most errors (up to 14 times IIRC, then it restarts from the beginning) but in this case it didn't know to treat HTTP 408 errors in that way so it just assumed it should give up right away and start over.  I'll make a change in the next release to let it resume on 408 errors like this so its less of a problem on unreliable networks.

**Comment by sabeechen** [comment]
I've released v0.104.1 of the addon which includes better handling of failures like this.
To update the addon, go to:
`Supervisor` > `Add-on Store` > Click the `...` in the top right > `Refresh`
and you should see the new version.

Kindly reopen this bug and let me know if  you continue to run into problems.
